src/Groups/GAPGroups.jl

```diff
@@ -575,19 +582,35 @@ normaliser = normalizer
 
 """
     core(G::Group, H::Group)
-Return `C,f`, where `C` is the normal core of `H` in `G`, and `f` is the embedding morphism of `C` into `G`.
+Return `C,f`, where `C` is the normal core of `H` in `G`,
+that is, the largest normal subgroup of `G` that is contained in `H`,
+and `f` is the embedding morphism of `C` into `G`.
 """
 core(G::T, H::T) where T<:GAPGroup = _as_subgroup(G, GAP.Globals.Core(G.X,H.X))
 
-# normal_closure(G::T, H::T) where T<:GAPGroup = T(GAP.Globals.NormalClosure(G.X,H.X))
-# we don't know how G,H embeds into N_C(G,H) 
+"""
+    normal_closure(G::Group, H::Group)
+
+Return `N, f`, where `N` is the normal closure of `H` in `G`,
+that is, the smallest normal subgroup of `G` that contains `H`,
+and `f` is the embedding morphism of `N` into `G`.
+
+Note that `H` must be a subgroup of `G`.
+"""
+normal_closure(G::T, H::T) where T<:GAPGroup = _as_subgroup(G, GAP.Globals.NormalClosure(G.X,H.X))
+
+# Note:
+# GAP admits `NormalClosure` also when `H` is not a subgroup of `G`,
+# and in this case the result is not contained in `G`.
+# (We should test whether `H` is a subgroup of `G`,
+# but then the user should have the possibility to omit this check.)
 
 """
     pcore(G::Group, p::Int64)
 Return `C,f`, where `C` is the `p`-core (i.e. the largest normal `p`-subgroup) of `G` and `f` is the embedding morphism of `C` into `G`.
 """
 function pcore(G::GAPGroup, p::Int64)
-   if !GAP.Globals.IsPrime(p)
+   if !GAP.Globals.IsPrimeInt(p)
```

Comment:
perhaps we should rather call the relevant function from Oscar/Nemo/Hecke/AbstractAlgebra

Reply:
Yes. (I had suggested this in the description.)